src/System.Buffers.Experimental/ReferenceCountedBuffer.cs

```diff
@@ -17,15 +17,16 @@ public override void Retain()
             Interlocked.Increment(ref _referenceCount);
         }
 
-        public override void Release()
+        public override bool Release()
         {
             if (!IsRetained) BuffersExperimentalThrowHelper.ThrowInvalidOperationException();
             if (Interlocked.Decrement(ref _referenceCount) <= 0) {
                 OnNoReferences();
             }
+            return IsRetained;
         }
 
-        public override bool IsRetained => _referenceCount > 0;
+        protected override bool IsRetained => _referenceCount > 0;
```

Comment:
This change isn't right. The idea was to return the result of the atomic operation.

```C#
public override bool Release()
{
  var result = Interlocked.Decrement(ref _referenceCount);
  if (result < 0) BuffersExperimentalThrowHelper.ThrowInvalidOperationException();
  if (result == 0) {
      OnNoReferences();
      return true;
  }
  return false;
}
```